### PR TITLE
fix: cannot destructure property finish details error

### LINKED ISSUE #895: [Bug] {{TypeError: Cannot destructure property 'finish_details' of 'E.message' as it is undefined.}}
### Please search before asking

- [X] I searched in the [issues](https://github.com/yetone/openai-translator/issues) and found nothing similar.

### Please read README

- [X] I have read the troubleshooting section in the README in detail.

### Please check your network and OpenAI API quota

- [X] I am sure it is not a network issue or an OpenAI API quota issue.

### OpenAI Translator version

0.1.23

### 系统/浏览器版本 System/Browser version

chrome 114.0.5735.198（正式版本） (arm64)

### 复现步骤 Reproduce step

昨天还可以正常使用， 直接划词后提示报错，免费版的chatGPT. 
已做检查
1. chatgpt 浏览器中可以正常使用
2. 插件更新了最新
3. 

### 你看到了什么错误？What errors do you see?

截屏2023-07-20 10 24 59

### 你期望看到什么？What did you expect to see?

正常使用

### 还有其他的内容吗？Anything else?

_No response_

### 你是否愿意提交一份 PR 来修改这个错误？Are you willing to submit a PR?

- [ ] 我愿意提供 PR! I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by linyimin0812**
fix #895

### CODE REVIEW

**Comment by maltoze** [comment]
image

**Comment by linyimin0812** [comment]
> 

应该是同时点击太多次了

image

File: src/common/translate.ts [hunk 1]
```diff
@@ -519,6 +519,11 @@ export async function translate(query: TranslateQuery) {
...
                     return
                 }
+
+                if (!resp.message) {
+                    return
+                }
+
                 const { finish_details: finishDetails } = resp.message
                 if (finishDetails) {
                     query.onFinish(finishDetails.type)
```
Comment (maltoze) [inline src/common/translate.ts:529]:
```suggestion
                    if (resp.is_completion) {
                        query.onFinish('stop')
                    }
```

Comment (linyimin0812) [reply src/common/translate.ts]: done

File: src/common/translate.ts [hunk 2]
```diff
@@ -519,9 +519,13 @@ export async function translate(query: TranslateQuery) {
...
                 }
-                const { finish_details: finishDetails } = resp.message
-                if (finishDetails) {
-                    query.onFinish(finishDetails.type)
+
+                if (!resp.message) {
+                    return
+                }
+
+                if (resp.is_completion) {
```
Comment (maltoze) [inline src/common/translate.ts:527]: data: {"message": {"id": "23feae01-d612-4423-93e1-8cda02f94cd4", "author": {"role": "assistant", "name": null, "metadata": {}}, "create_time": 1689829858.124457, "update_time": null, "content": {"content_type": "text", "parts": ["\u4fee\u590d\uff1a\u65e0\u6cd5\u89e3\u6784\u5c5e\u6027\u5b8c\u6210\u8be6\u60c5\u9519\u8bef\uff03897"]}, "status": "finished_successfully", "end_turn": true, "weight": 1.0, "metadata": {"finish_details": {"type": "stop", "stop": "<|diff_marker|>"}, "is_complete": true, "message_type": "next", "model_slug": "text-davinci-002-render-sha-mobile"}, "recipient": "all"}, "conversation_id": "79c91bcf-cbb5-40e8-92e3-4bdac412d052", "error": null}

data: {"conversation_id": "79c91bcf-cbb5-40e8-92e3-4bdac412d052", "message_id": "23feae01-d612-4423-93e1-8cda02f94cd4", "is_completion": true, "moderation_response": {"flagged": false, "blocked": false, "moderation_id": "modr-7eGEFYqhprvK90b5y8hGRGeefJq2E"}}

data: [DONE]

`is_completion` 为 `true` 时没有 `message` 字段

Comment (linyimin0812) [reply src/common/translate.ts:527]: data: {"conversation_id": "c66cdbb8-8d9e-453b-a02c-5c67d665930b", "message_id": "aaa2f89f-77db-4114-b4b3-523587181595", "is_completion": false, "moderation_response": {"flagged": false, "blocked": false, "moderation_id": "modr-7eEoqgZ9ywOe14AkJfhAZGe7f6QX1"}}

data: {"message": {"id": "3e61e37f-1dd9-41ab-838d-6f2c1d34b160", "author": {"role": "assistant", "name": null, "metadata": {}}, "create_time": 1689824440.902686, "update_time": null, "content": {"content_type": "text", "parts": ["\u8981"]}, "status": "in_progress", "end_turn": null, "weight": 1.0, "metadata": {"message_type": "next", "model_slug": "text-davinci-002-render-sha"}, "recipient": "all"}, "conversation_id": "c66cdbb8-8d9e-453b-a02c-5c67d665930b", "error": null}


存在`is_completion`为false的情况

Comment (linyimin0812) [reply src/common/translate.ts:527]: image
好像明白了，把判断is_completion放到前面了

**Review by maltoze — APPROVED** [review]

**Pull request merged**